pkg/virt-manifest/rest/manifest.go

```diff
@@ -0,0 +1,106 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	ghodss_yaml "github.com/ghodss/yaml"
+	"k8s.io/client-go/pkg/util/yaml"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+	"kubevirt.io/kubevirt/pkg/virt-manifest"
+)
+
+const (
+	MIME_YAML = "application/yaml"
+)
+
+type manifest struct {
+	libvirtUri    string
+	libvirtUser   string
+	libvirtPasswd string
+}
+
+type entityYAMLAccess struct {
+	ContentType string
+}
+
+// Read isn't currently used in this implementation
+func (e entityYAMLAccess) Read(req *restful.Request, v interface{}) error {
+	return fmt.Errorf("not implemented")
+}
+
+// Write marshals the value to YAML
+func (e entityYAMLAccess) Write(resp *restful.Response, status int, v interface{}) error {
+	data, err := ghodss_yaml.Marshal(v)
+	if err != nil {
+		return err
+	}
+	resp.WriteHeader(status)
+	resp.Write(data)
+	return nil
+}
+
+func ManifestService(libvirtUri string, libvirtUser string, libvirtPasswd string) (*restful.WebService, error) {
+	ws := new(restful.WebService)
+	m := manifest{
+		libvirtUri:    libvirtUri,
+		libvirtUser:   libvirtUser,
+		libvirtPasswd: libvirtPasswd}
+	restful.RegisterEntityAccessor(MIME_YAML, entityYAMLAccess{})
+	ws.Path("/").
+		Consumes(restful.MIME_JSON, MIME_YAML).
+		Produces(restful.MIME_JSON, MIME_YAML)
+	ws.Route(ws.GET("/api/v1/status").To(m.alive))
+	ws.Route(ws.POST("/api/v1/manifest").To(m.mapManifest))
+	return ws, nil
+}
+
+func (m manifest) alive(request *restful.Request, response *restful.Response) {
+	res := map[string]interface{}{"status": "ok"}
+	response.WriteEntity(res)
+}
+
+func (m manifest) mapManifest(request *restful.Request, response *restful.Response) {
+	log := logging.DefaultLogger()
+
+	vm := new(v1.VM)
+	err := yaml.NewYAMLOrJSONDecoder(request.Request.Body, 4096).Decode(vm)
+
+	if err != nil {
+		log.Error().Reason(err).Msg("Unable to decode VM spec")
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
+
+	if vm.ObjectMeta.Name == "" {
+		vm.ObjectMeta.Name = "unknown"
+	}
+	// Make sure the domain name matches the VM name
+	if vm.Spec.Domain == nil {
+		vm.Spec.Domain = new(v1.DomainSpec)
+	}
+	vm.Spec.Domain.Name = vm.ObjectMeta.Name
+	err = virt_manifest.AddMinimalDomainSpec(vm.Spec.Domain)
```

Comment:
I don't think that this should ever fail. It just fills in defaults if fields are empty.